File: 1807 D. Odd Queries/Solution.py
# Krishno_MD_KUET_CSE

t = int(input())

for i in range(t):
    n,q = map(int,input().split())
    v = list(map(int,input().split()))
    pr =[0]*n
    pr[0] = v[0]
    for i in range(1,n):
        pr[i] = pr[i-1] + v[i]
    for i in range(q):
        l,r,k = map(int,input().split())
        total = pr[n-1]
        ran_t = pr[r-1]- (pr[l-2] if l>1 else 0)
        mod = total-ran_t+(r-l+1)*k
        if mod%2 == 0:
            print("NO")
        else:
            print("YES")

File: 1807 D. Odd Queries/Solution.cpp
#include<bits/stdc++.h>
using namespace std; 

// Krishno_MD_KUET_CSE

void solve(){
    int n,q; cin>>n>>q;
    vector<int> v(n+1);
    vector<int> pr(n+1,0);
    for(int i=1; i<n+1; i++){
        cin>>v[i];
        pr[i] = pr[i-1] + v[i];

    }
    for(int i=0; i<q; i++){
        int l,r,k;
        cin>>l>>r>>k;
        int total = pr[n];
        int ran_t = pr[r] - pr[l-1];
        int mod = total - ran_t + (r-l+1)*k;
        if(mod%2 == 0){cout<<"NO\n";}
        else{cout<<"YES\n";}
    }
}

int main(){
    int t;
    cin >> t;
    while(t--){
        solve();
    }
    return 0;
}
